todo/pages/index.tsx

```diff
@@ -1,72 +1,207 @@
-            className={styles.card}
-          >
-            <h2>Deploy &rarr;</h2>
-            <p>
-              Instantly deploy your Next.js site to a public URL with Vercel.
-            </p>
-          </a>
-        </div>
-      </main>
+    const addTodo = () => {
+        const newTodo: Todo = {
+            id: nextId,
+            title: inputText,
+            editMode: false,
+            enabled: true,
+        }
+        setTodoList([...todoList, newTodo]);
+        setFlashMessages({ todoTitle: newTodo.title, message: 'が登録されました！' });
+        setInputText('');
+        nextId++;
+        console.log(todoList);
+    };
 
-      <footer className={styles.footer}>
-        <a
-          href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Powered by{' '}
-          <span className={styles.logo}>
-            <Image src="/vercel.svg" alt="Vercel Logo" width={72} height={16} />
-          </span>
-        </a>
-      </footer>
-    </div>
-  )
-}
+    const handleChangeEdit = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setEditText(e.target.value);
+    }
+
+    // 更新処理=======
+    const updateTodo = (id: number) => {
+        setTodoList((prevTodo) => prevTodo.map((todo) => todo.id === id ? { id: todo.id, title: editText, editMode: false, enabled: todo.enabled } : todo));
+        setFlashMessages({ todoTitle: editText, message: 'が更新されました！' });
+        setEditText('');
+    }
+
+    // 削除処理=======
+    const disableTodo = (id: number, title: string) => {
+        const todoTitle = title;
+        setTodoList(todoList.map((todo) => todo.id === id ? { id: todo.id, title: todo.title, editMode: todo.editMode, enabled: false } : todo));
+        setFlashMessages({ todoTitle: todoTitle, message: 'が削除されました！' });
+    };
+
+    return (
+        <div>
+            <Head>
+                <title>TODO</title>
+                <meta name="description" content="これはTODO作成アプリです。" />
+                <link rel="icon" href="/favicon.ico" />
+            </Head>
+            <header>
+                <p className={styles.title}>TODOリスト</p>
+            </header>
+            <main>
+                {buttonState === 'plus' ?
+                    <div>
+                        <Button
+                            onClick={enableMinusButton}
+                            className={styles.show}>追加
+                        </Button>
+                    </div>
+                    :
+                    <div>
+                        <Button
+                            onClick={enablePlusButton}
+                            className={styles.hide}>
+                            取消
+                        </Button>
+                    </div>
+                }
+                {(flashMessages) && <p>{flashMessages.todoTitle}{flashMessages.message}</p>}
```

Comment:
そもそもflashMessages内の変数を`message`だけにして、その中に`todoTitle + message`にした方が良さそう！

```suggestion
                {flashMessages && <p>{flashMessages.todoTitle}{flashMessages.message}</p>}
```